core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -814,13 +814,23 @@ public void tryReconnect() {
 
     @WebMethod(name="slave-agent.jnlp")
     public HttpResponse doSlaveAgentJnlp(StaplerRequest req, StaplerResponse res) {
-        return new EncryptedSlaveAgentJnlpFile(this, "slave-agent.jnlp.jelly", getName(), CONNECT);
+        return doJenkinsAgentJnlp(req, res);
+    }
+
+    @WebMethod(name="jenkins-agent.jnlp")
+    public HttpResponse doJenkinsAgentJnlp(StaplerRequest req, StaplerResponse res) {
```

Comment:
Unnecessary code duplication as `@WebMethod` takes a list of names.

If this is the future preferable API for this (even if unseen), should mark the old one `@deprecated`.

Also missing `@since` or `@Restricted`.